lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -341,74 +408,16 @@ func selectHopHints(amtMSat lnwire.MilliSatoshi, cfg *AddInvoiceConfig,
 			break
 		}
 
-		// Since we're only interested in our private channels, we'll
-		// skip public ones.
-		isPublic := channel.ChannelFlags&lnwire.FFAnnounceChannel != 0
-		if isPublic {
-			continue
-		}
-
-		// Make sure the counterparty has enough balance in the channel
-		// for our amount. We do this in order to reduce payment errors
-		// when attempting to use this channel as a hint.
-		chanPoint := lnwire.NewChanIDFromOutPoint(
-			&channel.FundingOutpoint,
+		// If the channel can'tbe a hop hint, then we'll skip it.
+		// Otherwise, we'll use the policy information to populate the
+		// hop hint.
+		remotePolicy, canBeHopHint := chanCanBeHopHint(
+			channel, graph, cfg,
 		)
-		if amtMSat >= channel.LocalCommitment.RemoteBalance {
-			log.Debugf("Skipping channel %v due to "+
-				"not having enough remote balance",
-				chanPoint)
-			continue
-		}
-
-		// Make sure the channel is active.
-		if !cfg.IsChannelActive(chanPoint) {
-			log.Debugf("Skipping channel %v due to not "+
-				"being eligible to forward payments",
-				chanPoint)
-			continue
-		}
-
-		// To ensure we don't leak unadvertised nodes, we'll make sure
-		// our counterparty is publicly advertised within the network.
-		// Otherwise, we'll end up leaking information about nodes that
-		// intend to stay unadvertised, like in the case of a node only
-		// having private channels.
-		var remotePub [33]byte
-		copy(remotePub[:], channel.IdentityPub.SerializeCompressed())
-		isRemoteNodePublic, err := graph.IsPublicNode(remotePub)
-		if err != nil {
-			log.Errorf("Unable to determine if node %x "+
-				"is advertised: %v", remotePub, err)
-			continue
-		}
-
-		if !isRemoteNodePublic {
-			log.Debugf("Skipping channel %v due to "+
-				"counterparty %x being unadvertised",
-				chanPoint, remotePub)
-			continue
-		}
-
-		// Fetch the policies for each end of the channel.
-		chanID := channel.ShortChanID().ToUint64()
-		info, p1, p2, err := graph.FetchChannelEdgesByID(chanID)
-		if err != nil {
-			log.Errorf("Unable to fetch the routing "+
-				"policies for the edges of the channel "+
-				"%v: %v", chanPoint, err)
+		if !canBeHopHint {
```

Comment:
nit: since we check `remotePolicy !=nil` below anyway, we could skip the bool here and just return a nil policy if it can't be a hint.